crates/ra_hir_ty/src/lower.rs

```diff
@@ -694,6 +673,56 @@ pub fn callable_item_sig(db: &dyn HirDatabase, def: CallableDef) -> PolyFnSig {
     }
 }
 
+pub fn associated_items<R>(
```

Comment:
we're really only interested in associated types here, so why not make it only iterate those, and pass `TypeAliasId`s?

also how about calling this `iterate_associated_type_shorthand_candidates` or something like that... though that's admittedly a bit long :thinking:

Reply:
>something like that... though that's admittedly a bit long thinking

something something when writing a command line compiler, you get tyctx, when writing an IDE, you get `iterate_associated_type_shorthand_candidates`

Reply:
:smile: I did think about adding, "... but we have completion"